README.md
zendserver Cookbook
====================
This cookbook installs and manages Zend Server

It allows bootstrapping Zend Server in single server mode or cluster mode.

Requirements
------------

#### Platforms
- Debian, Ubuntu
- RHEL, CentOS, Oracle Linux

#### Cookbooks
- apt
- yum

Attributes
----------

#### zendserver::single
<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['zendserver']['version']</tt></td>
    <td>string</td>
    <td>Zend Server version to install</td>
    <td><tt>6.1</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['phpversion']</tt></td>
    <td>string</td>
    <td>PHP version to install</td>
    <td><tt>5.4</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['ordernumber']</tt></td>
    <td>string</td>
    <td>The order number part of the license information</td>
    <td><tt>-</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['licensekey']</tt></td>
    <td>string</td>
    <td>The license key part of the license information</td>
    <td><tt>-</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['production']</tt></td>
    <td>boolean</td>
    <td>Bootstrap Zend Server in production (true)/development(false) mode (See zend server documentation for more details)</td>
    <td><tt>TRUE</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['apikeyname']</tt></td>
    <td>string</td>
    <td>Name fort the web API key that the installer creates. The api key is required for all management functionality</td>
    <td><tt>-</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['apikeysecret']</tt></td>
    <td>string</td>
    <td>A 64 character key used for signing API requests</td>
    <td><tt>-</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['adminemail']</tt></td>
    <td>string</td>
    <td>An email address for the Zend Server admin</td>
    <td><tt>-</tt></td>
  </tr>
</table>

#### zendserver::cluster
Requires all the attributes listed for single, plus a valid MySQL database.

<table>
  <tr>
    <th>Key</th>
    <th>Type</th>
    <th>Description</th>
    <th>Default</th>
  </tr>
  <tr>
    <td><tt>['zendserver']['dbhost']</tt></td>
    <td>string</td>
    <td>A MySQL database server host address - required for clustering</td>
    <td><tt>-</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['dbusername']</tt></td>
    <td>string</td>
    <td>The MySQL server username, must be able to create a database - required for clustering</td>
    <td><tt>-</tt></td>
  </tr>
  <tr>
    <td><tt>['zendserver']['dbpassword']</tt></td>
    <td>string</td>
    <td>The MySQL server password - required for clustering</td>
    <td><tt>-</tt></td>
  </tr>

</table>

Usage
-----
#### zendserver:single
Installs Zend Server, and bootstraps in single server mode

Just include `zendserver` in your node's `run_list`:

```json
{
  "name":"my_node",
  "run_list": [
    "recipe[zendserver::single]"
  ]
}
```

#### zendserver:cluster
Installs Zend Server, and bootstraps in cluster mode

Just include `zendserver` in your node's `run_list`:

```json
{
  "name":"my_node",
  "run_list": [
    "recipe[zendserver::cluster]"
  ]
}
```

Contributing
------------

e.g.
1. Fork the repository on Github
2. Create a named feature branch (like `add_component_x`)
3. Write you change
4. Write tests for your change (if applicable)
5. Run the tests, ensuring they all pass
6. Submit a Pull Request using Github

CHANGELOG.md
zend-server CHANGELOG
=====================

This file is used to list changes made in each version of the zend-server cookbook.

0.1.0
-----
- [your_name] - Initial release of zend-server

- - -
Check the [Markdown Syntax Guide](http://daringfireball.net/projects/markdown/syntax) for help with Markdown.

The [Github Flavored Markdown page](http://github.github.com/github-flavored-markdown/) describes the differences between markdown on github and standard markdown.


metadata.rb
name             'zendserver'
maintainer       'Zend Technologies'
maintainer_email 'maurice.k@zend.com'
license          'All rights reserved'
description      'Installs/Configures zendserver'
long_description IO.read(File.join(File.dirname(__FILE__), 'README.md'))
version          '0.8.0'

depends "apt"
depends "yum"
# depends "rightscale"

recipe "zendserver", "Adds the Zend Server repo and installs the package"
recipe "zendserver::single", "Install & bootstrap in 1 step"
recipe "zendserver::cluster", "Install, bootstrap and create/join a cluster in 1 step"
recipe "zendserver::manage", "Manages zend server, typically no direct actions"
recipe "zendserver::bootstrapsingle", "Bootstraps the server in single server mode"
recipe "zendserver::joincluster", "Join a Zend Server Cluster"

attribute "zendserver/version",
  :display_name => "Zend Server version",
  :description => "The version to install - must be a valid repository version",
  :required => "optional",
  :default => "6.1",
  :recipes => [ 
      "zendserver::install"
  ]

attribute "zendserver/phpversion",
  :display_name => "PHP version",
  :description => "The PHP version to install - must be a valid PHP version for the Zend Server selected version",
  :required => "optional",
  :default => "5.4",
  :recipes => [ 
      "zendserver::install"
  ]

attribute "zendserver/adminpassword",
  :display_name => "Admin password",
  :description  => "Zend Server GUI admin password",
  :required	=> "required",
  :recipes		=> [
  	"zendserver::bootstrapsingle"
  ]

attribute "zendserver/ordernumber",
  :display_name => "Order number",
  :description  => "Zend Server license order number",
  :required	=> "required",
  :recipes		=> [
  	"zendserver::bootstrapsingle"
  ]

attribute "zendserver/licensekey",
  :display_name => "License key",
  :description  => "Zend Server license key",
  :required	=> "required",
  :recipes		=> [
  	"zendserver::bootstrapsingle"
  ]

attribute "zendserver/production",
  :display_name => "Production mode",
  :description  => "Whether to set Zend Server up as a development or production server (Monitoring rules, agregation, ...).",
  :required	=> "optional",
  :choice => ["TRUE", "FALSE"],
  :default  => "TRUE",
  :recipes		=> [
  	"zendserver::bootstrapsingle"
  ]

attribute "zendserver/apikeyname",
  :display_name => "Api key name",
  :description  => "An API key we can subsequently use",
  :required	=> "optional",
  :default  => "rightscale",
  :recipes		=> [
  	"zendserver::bootstrapsingle",
    "zendserver::manage"
  ]

attribute "zendserver/apikeysecret",
  :display_name => "Api key secret",
  :description  => "An API key secret we can sbsequenly use (64 chars)",
  :required	=> "required",
  :recipes		=> [
  	"zendserver::bootstrapsingle",
    "zendserver::manage"
  ]

attribute "zendserver/adminemail",
  :display_name => "Admin Email",
  :description  => "Administrator's email (to send alerts to)",
  :required	=> "optional",
  :recipes		=> [
  	"zendserver::bootstrapsingle"
  ]

attribute "zendserver/devpassword",
  :display_name => "Developer password",
  :description  => "Developer's account password",
  :required	=> "optional",
  :recipes		=> [
  	"zendserver::bootstrapsingle"
  ]

attribute "zendserver/dbhost",
  :display_name => "Database Host name/IP",
  :description  => "A valid MySQL database for the clustering functionality.",
  :required => "optional",
  :recipes    => [
    "zendserver::joincluster"
  ]

attribute "zendserver/dbusername",
  :display_name => "MySQL username",
  :description  => "The MySQL username",
  :required => "optional",
  :recipes    => [
    "zendserver::joincluster"
  ]

attribute "zendserver/dbpassword",
  :display_name => "MySQL password",
  :description  => "The MySQL password",
  :required => "optional",
  :recipes    => [
    "zendserver::joincluster"
  ]